sdcm/sct_events.py

```diff
@@ -331,30 +388,117 @@ def run(self):
         LOGGER.info("writing to %s", self.events_filename)
 
         for event_type, message_data in EVENTS_PROCESS.subscribe_events():
+            msg = "{}: {}".format(message_data.formatted_timestamp(), str(message_data).strip())
             with open(self.events_filename, 'a+') as log_file:
-                log_file.write(str(message_data).strip() + '\n')
-            LOGGER.info(message_data)
+                log_file.write(msg + '\n')
+            LOGGER.info(msg)
+
+
+class PrometheusDumper(threading.Thread):
+    def __init__(self):
+        self.stop_event = threading.Event()
+        super(PrometheusDumper, self).__init__()
+
+    def run(self):
+        events_gauge = nemesis_metrics_obj().create_gauge('sct_events_gauge',
+                                                          'Gauge for sct events',
+                                                          ['event_type', 'type', 'severity', 'node'])
+
+        for event_type, message_data in EVENTS_PROCESS.subscribe_events(stop_event=self.stop_event):
+            events_gauge.labels(event_type,
+                                getattr(message_data, 'type', ''),
+                                message_data.severity,
+                                getattr(message_data, 'node', '')).set(message_data.timestamp)
+
+    def terminate(self):
+        self.stop_event.set()
+
+
+class GrafanaAnnotator(threading.Thread):
+    def __init__(self, specific_dashboard=False):
+        self.stop_event = threading.Event()
+        self.url_set = threading.Event()
+        self.grafana_base_url = ''
+        self.specific_dashboard = specific_dashboard
+        self.dashboard_id = None
+        self.panel_id = None
+
+        self.auth = ('admin', 'admin')
+        super(GrafanaAnnotator, self).__init__()
+
+    def set_grafana_url(self, grafana_base_url):
+        self.grafana_base_url = grafana_base_url
+        if self.specific_dashboard:
+            res = requests.get(self.grafana_base_url + '/api/search', params={'query': 'nemesis'}, auth=self.auth).json()
```

Comment:
separate to functions and remove from the `set_grafana_url` flow

Reply:
done